src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        addLastEntry(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        if (index == 0) {
+            addFirstEntry(value);
+        } else if (index == size) {
+            addLastEntry(value);
+        } else {
+            addByIndexEntry(value, index);
+        }
```

Comment:
this is the only usage of checkIndex with the second argument being size + 1. If we could use here the usual `size`, we could eliminate this second argument from the method signature, think about where to place it so we can use the usual check to the same effect